lib/collection/src/shards/forward_proxy_shard.rs

```diff
@@ -305,7 +305,7 @@ impl ShardOperation for ForwardProxyShard {
             let tick = result.clock_tag.map(|tag| tag.clock_tick);
             let remote_tick = remote_result.clock_tag.map(|tag| tag.clock_tick);
 
-            if remote_tick > tick || tick.is_none() {
```

Comment:
why is this change fine?

Reply:
The simplified condition behaves slightly different on inputs:

| `remote_tick` | `tick`   | before | after |
|---------------|----------|------  |-------|
| `None`          | `None`     | true   | false |
| `Some`          | `None`     | true   | true  |
| `None`          | `Some`     | false  | false |
| `Some(1)`       | `Some(2)`  | false  | false |
| `Some(2)`       | `Some(1)`  | true   | true  |
| `Some(1)`       | `Some(1)`  | false  | false |

There's one discrepancy as you can see. However, in that case the clock tag itself will be exactly the same, and so the end result is exactly the same.

Even though it's fine, I'll revert my change because the previous version may be a bit easier to understand.